clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -645,6 +647,11 @@ private Collection<SelectionKey> determineHandlingOrder(Set<SelectionKey> select
     private void attemptRead(SelectionKey key, KafkaChannel channel) throws IOException {
         //if channel is ready and has bytes to read from socket or buffer, and has no
         //previous receive(s) already staged or otherwise in progress then read from it
+        log.trace("attempt read from the channel (channel.ready:{}, key.isReadable:{}, " +
```

Comment:
should we also print the channel id to identify the src/dst host?

Reply:
Good suggestion. It's added.